go-controller/pkg/ovn/policy.go

```diff
@@ -853,6 +854,14 @@ func (oc *Controller) handlePeerNamespaceAndPodSelector(
 				np.podHandlerList = append(np.podHandlerList, podHandler)
 			},
 			DeleteFunc: func(obj interface{}) {
+				// when the namespace labels no longer apply
+				// remove the namespaces pods from the address_set
+				namespace := obj.(*kapi.Namespace)
+				pods, _ := oc.kube.GetPods(namespace.Name, *podSelector)
+				for _, pod := range pods.Items {
+					oc.handlePeerPodSelectorDelete(gp, &pod)
+				}
+
 			},
 			UpdateFunc: func(oldObj, newObj interface{}) {
```

Comment:
Should we also handle the "update label" case as well?

Reply:
we don't have to worry about the update case because of how `AddFilteredNamespaceHandler()` works when a Namespace is updated so that the filter no longer applies it comes to the filteredNamespaceHandler as a delete event. Since we only care about when namespaces get added or in this case removed from the networkPolicy we only have to worry about the delete case